### ISSUE TITLE
Database does not save entites

### ISSUE THREAD

**Issue by nays12** [nays12/QRCode-Authentication-Web#17]
For some reason, whenever I call `dbconn.saveChanges();` anywhere in the project. I get this error. Everything was working fine yesterday and I have not changed anything since so I am not sure what this is but I will try and fix it.

![image](https://user-images.githubusercontent.com/41583144/69020764-4c108580-097b-11ea-9005-57db451043e7.png)

**Comment by nays12** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Set Digital Credentials](https://trello.com/c/d37M2ePg/46-set-digital-credentials)

**Comment by nays12** [comment]
This stack overflow helped with the migrations

https://stackoverflow.com/questions/15653674/update-database-column-does-not-exist

This Microsoft documentation and other links helped with the Fluent API definitions:

https://docs.microsoft.com/en-us/ef/core/modeling/relationships
https://www.learnentityframeworkcore.com/configuration/fluent-api/hasmany-method
https://stackoverflow.com/questions/21885706/multiple-one-to-many-relations-of-the-same-entity-type-code-first

**Comment by nays12** [comment]
So I updated the one to many relationship and this is the new error 🙄

![image](https://user-images.githubusercontent.com/41583144/69024154-2d17f080-0987-11ea-8109-e2e555f6746c.png)

**Comment by nays12** [comment]
Finally enabled migrations. New error

![image](https://user-images.githubusercontent.com/41583144/69028826-cdc2dc00-0998-11ea-8c6d-d26a8f248a1a.png)

**Comment by nays12** [comment]
Interesting new error

![image](https://user-images.githubusercontent.com/41583144/69029253-0b743480-099a-11ea-8ef6-29fe1f627069.png)

**Comment by nays12** [comment]
I'm not entirely sure if I understand correctly what I fixed but I just cloned a previous version of our project, deleted the migrations and tables and created a new DbContext, migrations, dbsets, everything basically and I still had the same issue. I was getting the ambiguous column error again, and what I think fixed it was I went to the SQL Server Object Explorer and deleted the dbo._Migrations table and then everything finally recreated as I had wanted!

Sooo I think Entity was looking at that table and seeing the old configurations and referencing them .

Nevertheless, glad this is resolved! I merged the database-refactor branch into set-credentials and tested some CRUD functions and it works again.
